lib/schemas.js

```diff
@@ -103,7 +103,7 @@ function normalizeSchema (routeSchemas, serverOptions) {
       }
 
       if (!hasContentMultipleContentTypes) {
-        routeSchemas.response[code] = getSchemaAnyway(routeSchemas.response[code], serverOptions.jsonShorthand)
+        routeSchemas.response[code] = getSchemaAnyway(routeSchemas.response[code], serverOptions.jsonShorthand, code === '204')
```

Comment:
I would follow a different path here.

I would ignore the schema for this point:
https://github.com/fastify/fastify/blob/fd62068892b7312fcb8552834fbf60a035442ad3/lib/schemas.js#L154

I think it worth to say "if the standard does not expect a payload and the object does not contain the properties, just ignore it"

Reply:
Maybe you meant to mention the `getSchemaAnyway` function?